src/components/ui/coaching-session-selector.tsx

```diff
@@ -56,7 +63,7 @@ function CoachingSessionsSelectItems({
       try {
         const sessionIds = coachingSessions?.map((session) => session.id) || [];
         const goalsPromises = sessionIds.map((id) =>
-          fetchOverarchingGoalsByCoachingSessionId(id)
+          OverarchingGoalApi.list(id)
```

Comment:
This component and others should ideally be updated to use hooks instead of the direct function calls, but I decided it was out of the scope of this PR to refactor to this level since it would be a pretty major component code restructuring change.